source/SharpGL/Core/SharpGL/RenderContextProviders/DIBSectionRenderContextProvider.cs

```diff
@@ -84,15 +84,18 @@ public override void SetDimensions(int width, int height)
 
 	    public override void Blit(IntPtr hdc) 
 	    {
+            // [RS] Because of this warning, we remove the if-statement.
+            // Warning 1: The result of the expression is always 'true' since a value of type 'System.IntPtr' is never equal to 'null' of type 'System.IntPtr?'	F:\Git Repositories\SharpGL\source\SharpGL\Core\SharpGL\RenderContextProviders\DIBSectionRenderContextProvider.cs	88	17	SharpGL
+
             //  We must have a device context.
-            if (deviceContextHandle != null)

```

Comment:
Correct, this code is totally wrong. What it should be is:

```cs
if (deviceContextHandle == IntPtr.Zero)
{
    return;
}
// not 
// if (deviceContextHandle != null)
```

As there is no point swapping buffers if the dc is null. Also, inverting the logic means less indentation for the 'real work' of the code